htdocs/core/db/DoliDB.class.php

```diff
@@ -459,4 +459,25 @@ public function getRows($sql)
 
 		return false;
 	}
+
+	/**
+	 * Prepare a SQL statement for execution
+	 *
+	 * @param string $sql SQL query to prepare
+	 * @return false|mysqli_stmt
+	 */
+	public function prepare($sql)
+	{
+		if (!$this->connected) {
+			$this->lasterror = 'Not connected to database';
+			return false;
+		}
+
+		$stmt = $this->db->prepare($sql);
```

Comment:
I think the implementation with
$stmt = $this->db->prepare($sql);

should not be into the DoliDB.class.php but into the DoliDBMysqli class. Because method ->prepare does not exists in postgresql.
So we can add the 
public function prepare($sql)
with empty content into DoliDB.class.php
and the complete implementation into class  DoliDBMysqli  and DoliDBPgsql